### PR TITLE
Use resource_prefix instead of namespace in K8s tests

### PR DESCRIPTION
**Pull request by lidizheng**
This PR fixes 2 usability issues:

1. Namespace is accepted though not actually used;
2. Chicken-and-egg problem with loading default backend service's name before creating it.

[sponge/04f22bae-2a82-48b4-a90c-5fc7e84fb08e](http://sponge/04f22bae-2a82-48b4-a90c-5fc7e84fb08e)

### CODE REVIEW

**Review by menghanl — APPROVED** [review]

File: tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s.py
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s.py:310]: this is an actual kubectl flag

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py:275]: This change is good. If you want this in, let's move it to another PR?

**Review by sergiitk — CHANGES_REQUESTED** [review]
This is blocked on migration staging tests. Now that staging is broken, we can't migrate it.

File: tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s.py
Comment (lidizheng) [reply tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s.py]: Good catch.

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py
Comment (lidizheng) [reply tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py]: Created as https://github.com/grpc/grpc/pull/26713

**Comment by lidizheng** [comment]
@sergiitk Can you ping this PR once the staging migration is ready?

**Comment by sergiitk** [comment]
@lidizheng Yes, once it's done, I'll come back and approve this PR.

File: tools/run_tests/xds_k8s_test_driver/config/local-dev.cfg.example
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/config/local-dev.cfg.example:15]: Let's offer a value here?

```suggestion
--resource_prefix=xds-k8s-test
```

Ideally this should be usable out-of-box, by simply copying the example file. Or with minimal updates described in README.txt.

**Review by sergiitk — APPROVED** [review]
LGTM. Better to wait with merging it until tomorrow.

**Comment by lidizheng** [comment]
All psm-sec tests are passing in Testgrid. Merging.

**Pull request merged**
